### PR TITLE
Matching brackets

### PR DESCRIPTION
**Pull request by CorneliaKelinske**
Just a quick review please! Before I submit this to the brand new exercism site

### CODE REVIEW

File: elixir/matching-brackets/lib/matching_brackets.ex [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  @doc """
+  Checks that all the brackets and braces in the string are matched correctly, and nested correctly
+  """
+  @matching_brackets ["()", "[]", "{}"]
+
+  def check_brackets("") do
+    true
+  end
+
+  @spec check_brackets(String.t()) :: boolean
```
Comment (alanvardy) [inline elixir/matching-brackets/lib/matching_brackets.ex:11]: You may wish to move this spec to above the first function head.

File: elixir/matching-brackets/lib/matching_brackets.ex [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  def check_brackets(str) do
+    str = Regex.replace(~r/[^(){}\[\]]/, str, "")
```
Comment (alanvardy) [inline elixir/matching-brackets/lib/matching_brackets.ex:13]: I don't understand this regex. It seems to just handle a very specific case. Is it needed?

**Review by alanvardy — APPROVED** [review]

File: elixir/matching-brackets/lib/matching_brackets.ex (hunk 2 above)
Comment (CorneliaKelinske) [reply elixir/matching-brackets/lib/matching_brackets.ex:13]: Yes, it replaces everyting that is not a bracket.
In order to run my string replace function I need to get rid of all the clutter. In a string like: "\\left(\\begin{array}{cc} \\frac{1}{3} & x\\\\ \\mathrm{e}^{x} &... x^2 \\end{array}\\right)", my strategy wouldn't work.

Comment (alanvardy) [reply elixir/matching-brackets/lib/matching_brackets.ex:13]: Oh cool! I read it wrong and thought that the ^ meant start of line for some reason!

**Pull request merged**